### ISSUE TITLE
Discriminates &&= vs ... && ... =

### ISSUE THREAD

**Issue by forthrin** [rubocop/rubocop#14788]
## Steps to reproduce the problem

```ruby
def a(bar)
  foo = 'HOCKEY' # Lint/UselessAssignment
  bar &&= (
    foo = 'STICKS'
    bar * 10
  )
  pp [foo, bar]
end

def b(bar)
  foo = 'HOCKEY'
  bar && bar = (
    foo = 'STICKS'
    bar * 10
  )
  pp [foo, bar]
end

a(nil)
a(1)
b(nil)
b(1)
```

## Expected behavior

"You're free to go."

## Actual behavior

"Say, that's a nice bike."

## RuboCop version

```
System Version: macOS 26.1 (25B78)
Model Identifier: MacBookAir10,1
ruby 4.0.1 (2026-01-13 revision e04267a14b) +PRISM [arm64-darwin25]
rubocop     2026-01-16 2f516e2f5
```
